pixeltable/catalog/catalog.py

```diff
@@ -727,11 +910,88 @@ def _load_tbl_version(self, tbl_id: UUID, effective_version: Optional[int]) -> O
             )
 
         row = conn.execute(q).one_or_none()
-        tbl_record, schema_version_record = _unpack_row(row, [schema.Table, schema.TableSchemaVersion])
+        assert row is not None, f'Table record not found: {tbl_id}:{effective_version}'
+        tbl_record, version_record, schema_version_record = _unpack_row(
+            row, [schema.Table, schema.TableVersion, schema.TableSchemaVersion]
+        )
+        assert tbl_record.id == tbl_id
         tbl_md = schema.md_from_dict(schema.TableMd, tbl_record.md)
+        version_md = schema.md_from_dict(schema.TableVersionMd, version_record.md)
         schema_version_md = schema.md_from_dict(schema.TableSchemaVersionMd, schema_version_record.md)
+
+        return schema.FullTableMd(tbl_md, version_md, schema_version_md)
+
+    def store_tbl_md(
+        self,
+        tbl_id: UUID,
+        tbl_md: Optional[schema.TableMd],
+        version_md: Optional[schema.TableVersionMd],
+        schema_version_md: Optional[schema.TableSchemaVersionMd],
+    ) -> None:
+        """
+        Stores metadata to the DB. If specified, `tbl_md` will be updated in place (only one such record can exist
+        per UUID); `version_md` and `schema_version_md` will be inserted as new records.
+
+        If inserting `version_md` or `schema_version_md` would be a primary key violation, an exception will be raised.
+        """
+        conn = Env.get().conn
+
+        if tbl_md is not None:
+            conn.execute(
```

Comment:
Check the returned status to verify that this updated a row.

Reply:
Done